### ISSUE TITLE
How to get builder model index ?

### ISSUE THREAD

**Issue by danyj** [ThemeFuse/Unyson#524]
I know you busy but I am also stuck waiting on few fixes and not coming any further. 
Can you ad least help me on this 

http://stackoverflow.com/questions/30087770/backbone-underscore-how-to-get-index-of-a-model-inside-render

**Comment by moldcraft** [comment]
```javascript
render: function() {
    var index = this.model.collection.indexOf(this.model);
```

Note: I haven't tested this code

**Comment by danyj** [comment]
collection is not there bud

**Comment by danyj** [comment]
and I think who ever made that app forgot

    _.bindAll(this, “render”);


according to this 
https://bardevblog.wordpress.com/2012/01/16/understanding-backbone-js-simple-example/


"  If we did not include “_.bindAll(this, “render”);” in initialize then “this” in the render function would not be valid.  "

**Comment by danyj** [comment]
@moldcraft 

unbelievable! , almost 2 days digging trough code , 0.012 delay kills everything 
https://github.com/ThemeFuse/Unyson-Builder-Extension/blob/master/includes/option-types/builder/static/js/builder.js#L350

finally found it. ok ,  now this one is bit trickier and I know you not that well with backbone but this is your modal so you might be able to help 


now I have index and naming my grids as G1 G2....  


http://prntscr.com/72jq84

but I need to update the modal hidden option of that particular section ( grid ) 


I tried 


    fw.ops('section_name',index, self.view.model.get('options'));

the thing is that is updating all   options in a collection with a name   of  section_name
and I need to update that one for each option set. 
http://prntscr.com/72jqxv

**Comment by moldcraft** [comment]
>     fw.ops('section_name',index, self.view.model.get('options'));

Do not update the Backbone model attributes directly (by reference). You must use `model.set('attribute', value)` for proper 'change' events trigger (on `change` the hidden input is updated).

```javascript
var options = _.clone(self.view.model.get('options'));
fw.ops('section_name', index, options);
self.view.model.set('options', options);
```

*I hope, I understood you correctly.*

**Comment by danyj** [comment]
that is it!  thank you!!!!!!!!!!

**Comment by danyj** [comment]
Just opening this for a sec instead of making new issue. 


Is there an event I can hook on to when item is dragged from one collection to another? 

For example    column dragged from one section to another. 
http://prntscr.com/72u14z

How can I detect that event?
I need to check /fix  name indexes on that event for example when if you have 2 sections and in each you have 

Column 1  ,  Column 2 


when Column 1 from first section  is dropped in section that already has  Column 1  ,  Column 2 
I need to update the name. 

hope that this did not confuse you much

**Comment by moldcraft** [comment]
Look for Collection events http://backbonejs.org when a model is added to the collection

**Comment by danyj** [comment]
ok , will digg that way , thnx!
